src/firecracker/src/main.rs

```diff
@@ -255,6 +262,22 @@ fn main() {
     }
 }
 
+// Print supported snapshot data format versions.
+fn print_supported_snapshot_versions() {
+    print!("Supported snapshot data format versions:");
+    let mut snapshot_versions: Vec<String> = FC_VERSION_TO_SNAP_VERSION
+        .iter()
+        .map(|(key, _)| key.clone())
+        .collect();
+    snapshot_versions.sort();
+
+    let len = snapshot_versions.len();
+    for version in snapshot_versions.iter().take(len - 1) {
+        print!(" v{},", version);
+    }
+    println!(" v{}\n", snapshot_versions[len - 1]);
```

Comment:
If I understand correctly, the only reason this `for` loop takes the first `n-1` elements is because we want to have a newline after the last printed version. 

Wouldn't this be simpler?

```suggestion
    snapshot_versions.sort().iter().for_each(|version| print!(" v{},", version));
    println!("\n");
```

Reply:
I also wanted to avoid having a trailing comma at the end of the versions list.

Reply:
I see. In this case we could build a string and pop off the last element right before outputting. Anyway, this is not a huge deal, but I think that it would make for more readable code. it's up to you

Reply:
Comment addressed.